src/build.ts

```diff
@@ -0,0 +1,186 @@
+import { ensureDir, getProtocolInterface, parse, path } from "./deps.ts";
+import type { Protocol } from "./deps.ts";
+import { cleanManifest, createManifest } from "./get_manifest.ts";
+
+export const validateAndCreateFunctions = async (
+  workingDirectory: string,
+  outputDirectory: string,
+  // deno-lint-ignore no-explicit-any
+  manifest: any,
+  hookCLI: Protocol,
+) => {
+  // Clean output dir prior to build
+  if (await removeDirectory(outputDirectory)) {
+    hookCLI.log(`removed directory: ${outputDirectory}`);
+  }
+  // Ensure functions directory exists
+  const functionsPath = path.join(outputDirectory, "functions");
+  await ensureDir(functionsPath);
+
+  // Find all the run on slack functions
+  for (const fnId in manifest.functions) {
+    const fnDef = manifest.functions[fnId];
+
+    // For now we'll bundle all functions until this is available on a manifest
+    // TODO: add this check back once we add it to the manifest definition
+    // if (fnDef.runtime_environment !== 'slack') {
+    //   continue;
+    // }
```

Comment:
I believe we're now setting this — might be able to remove the comment, but I'm okay if we don't add that on to this PR, would be nice to combine with the below line checking for `type: "API"`

Reply:
Yeah let's not add scope creep to this PR, which is already creeping as it is 😅 
Agree with you, that'd be nice. Can you elaborate on how we can tackle this TODO? Are these properties set by deno-slack-sdk? I can tackle this as a follow-up.